### ISSUE TITLE
Difference between labels and annotations

### ISSUE THREAD

**Issue by arpitjain305** [OpenSLO/OpenSLO#213]
I was wondering how to interpret labels and annotations in the OpenSLO spec. For labels, I understand it is basically metadata that better describes the SLO, and enables me to classify it into some logical groupings (at least conceptually). Some examples of labels I can think of:
`team = dev`, `cloud = aws` etc

How to think about annotations in a similar manner? What are some examples?

**Comment by nieomylnieja** [comment]
hey @arpitjain305, while both of these can convey similar details, there's a major distinction between them in terms of the purpose they serve:

- labels are your custom metadata for each OpenSLO object, these should be respected by vendors (persisted when converting to vendor specific spec) and help you define custom grouping of resources, as you mentioned with your examples
- annotations on the other hand allow vendor specific instructions to be conveyed, they will be (most likely) discarded during vendor spec conversion too. Let's say the vendor extends `DataSource.metadata` definition with `project` field, to convey that information the vendor might utilise an annotation:

```yaml
apiVersion: openslo/v1
kind: DataSource
metadata:
  name: my-cloudwatch
  labels:
    team: dev
  annotations:
    vendor_xxx/project: default
spec:
  type: CloudWatch
  connectionDetails:
    accessKeyID: accessKey
    secretAccessKey: secretAccessKey
```

as a result we might get:
```yaml
apiVersion: vendor_xxx/v1
kind: DataSource
metadata:
  name: my-cloudwatch
  labels:
    team: dev
  project: default
spec:
  type: CloudWatch
  connectionDetails:
    accessKeyID: accessKey
    secretAccessKey: secretAccessKey
```

Tldr: annotations are a workaround for vendor specific logic for conversion from OpenSLO, I'm personally not a fan of that solution as it makes your OpenSLO spec, well... vendor specific :) Maybe in time, as a community, we could revisit annotations purpose :shrug:

**Comment by arpitjain305** [comment]
This is really helpful, thank you!
